module/VuFind/tests/unit-tests/src/VuFindTest/Search/Solr/ResultsTest.php

```diff
@@ -107,6 +113,230 @@ public function testGetResultTotal(): void
         $this->assertEquals(5, $results->getResultTotal());
     }
 
+    /**
+     * Test retrieving facets.
+     *
+     * @return void
+     */
+    public function testGetFacetList(): void
+    {
+        $config = $this->getMockBuilder(PluginManager::class)
```

Comment:
Do you think it's worth creating a ConfigPluginManagerTrait that includes a more generalized version of this code for mocking out the config manager?

https://github.com/vufind-org/vufind/blob/dev/module/VuFind/tests/unit-tests/src/VuFindTest/Search/Summon/ParamsTest.php#L53-L68

I'm envisioning a method that accepts an associative array of config filename => config array, and which uses a callback to return an appropriate config object (defaulting to empty for undefined values). This seems generally useful, and it would be more targeted than accepting `$this->any()` here (not to mention making individual tests shorter and more readable).

I'd be willing to do some refactoring as a subsequent pass if you think it's worth doing but don't have the time.

Reply:
That's a very good idea, now added.